File: Data Structures & Algorithms/insert-new-interval/submission-3.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> res = new ArrayList<>();

        for (int[] interval: intervals) {
            int x = interval[0];
            int y = interval[1];

            if (newInterval[0] > y) {
                res.add(new int[]{x, y});
            } else if (newInterval[1] < x) {
                res.add(newInterval);
                newInterval = new int[]{x, y};
            } else {
                newInterval[0] = Math.min(x, newInterval[0]);
                newInterval[1] = Math.max(y, newInterval[1]);
            }
        }
        res.add(newInterval);
        return res.toArray(new int[res.size()][]);
    }
}

File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        max_area = 0
        l, r = 0, len(heights)-1
        while l < r:
            area = min(heights[l], heights[r]) * (r-l)
            if heights[l] < heights[r]:
                l += 1
            else:
                r -= 1
            max_area = max(max_area, area)
        
        return max_area